python/ray/data/tests/block_batching/test_util.py

```diff
@@ -131,7 +134,39 @@ def gen(base_iterator):
     assert e.match("Fail")
 
 
-logger = logging.getLogger(__file__)
+@pytest.mark.parametrize("buffer_size", [0, 1, 2])
+def test_make_async_gen_varying_seq_lengths(buffer_size: int):
+    """Tests that iterators of varying lengths are handled appropriately"""
+
+    def _gen(base_iterator):
+        worker_id = base_iterator.worker_id
+
+        # Make workers produce sequences increasing the same order
+        # as worker-ids (so that for left workers sequences run out first)
+        target_length = worker_id + 1
+
+        return iter([f"worker_{worker_id}:{i}" for i in range(target_length)])
```

Comment:
weird to add the _WorkerIterator only for tests. 
Can we just use the next item value to determine  the target length?

Reply:
Good call